src/MongoDB.EntityFrameworkCore/Query/Visitors/ProjectionBindingRemovingExpressionVisitorBase.cs

```diff
@@ -266,7 +266,7 @@ protected override Expression VisitMethodCall(MethodCallExpression methodCallExp
         IProperty property,
         Type type)
     {
-        var storeName = property.GetElementName();
+        var storeName = property.GetDefaultElementName();// .GetElementName();
         if (storeName.Length == 0)
```

Comment:
If you are OK with suggestion above - we should refactor this by moving the conditions from GetDefaultElementName extension method to here.